### DISCUSSION TITLE
Check for external volume change

### DISCUSSION THREAD

**Discussion by DementevAlex** [librespot-org/librespot#611] [Feature Requests]
Hello!
Just got librespot to control volume through the ALSA mixer with no problems.
I also have IR receiver attached to my Raspberry PI to do some stuff including volume control. It also works with ALSA mixer.

And as I understand it seems like librespot doesn't check the volume value. For example I set volume on my phone to 60% then changed it to 30% via IR. After that I still see 60% on my phone. 

Is it possible to add the volume value reading?

**Comment by ashthespy** [comment]
This is on my [#todo list](https://github.com/librespot-org/librespot/pull/443#issuecomment-596521336). The bits and bobs are there, I hacked up a solution some time ago, but it's a waste of resources, as it spins up a thread just to poll for volume changes..

**Comment by DementevAlex** [comment]
Well that's good news, thanks.
After some googling I found the spotcontrol. If I can set up lirc to control volume through it (not through ALSA directly) and volume level will be shown on my phone or desktop app then it would fit me as a tmp solution.
But it's going to be even more waste of resources :)

**Comment by celismax** [comment]
Hi @AlexanderShevelev !
Did you find a workaround unsing ALSA ?
